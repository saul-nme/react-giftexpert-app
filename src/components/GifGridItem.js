import React from 'react'

export default function GifGridItem({id, title, url}) {
   return (
      <div className="card animate__animated animate__pulse">
         <img src={url} alt={title} />
         <p>{title}</p>
      </div>
   )
}
