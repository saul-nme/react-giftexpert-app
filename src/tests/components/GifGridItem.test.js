import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {

   const title = "coso"
   const url = "https://localhost/coso.png"
   const wrapper = shallow(<GifGridItem title={title} url={url} />);

   test('Debe mostrar el componente correctamente', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('debe de tener un pÃ¡rrafo cone el title', () => {
      const p = wrapper.find('p');
      expect(p.text().trim()).toBe(title);
   });

   test('debe tener url y alt de los props', () => {
      const img = wrapper.find('img');
      expect(img.prop('src')).toBe(url);
      expect(img.prop('alt')).toBe(title);
   });

   test('debe tenet animate__pulse', () => {
      const card = wrapper.find('.card');
      expect(String(card.prop('className')).indexOf("animate__pulse") >= 0).toBe(true)
   });

});
