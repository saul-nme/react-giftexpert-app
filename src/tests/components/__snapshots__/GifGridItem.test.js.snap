// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__pulse"
>
  <img
    alt="coso"
    src="https://localhost/coso.png"
  />
  <p>
    coso
  </p>
</div>
`;
